File: src/provider/PluginAvailabilityProvider.java
package provider;

import com.intellij.psi.PsiElement;
import factory.PomFactory;

/**
 *
 */
public class PluginAvailabilityProvider {
    private final PomFactory pomFactory = new PomFactory();

    /**
     *
     * @param psiElement
     * @return
     */
    public boolean provide(final PsiElement psiElement) {
        return pomFactory.create(psiElement) != null;
    }
}

File: src/main/kotlin/provider/PluginAvailabilityProvider.kt
package provider

import com.intellij.psi.PsiElement
import factory.PomFactory

class PluginAvailabilityProvider(
        private val pomFactory: PomFactory = PomFactory(),
        private val freeVersionProvider: FreeVersionProvider = FreeVersionProvider()
) {
    fun provide(psiElement: PsiElement): Boolean {
        val pom = pomFactory.create(psiElement)
        return if (pom != null) {
            freeVersionProvider.getFreeVersion(pom.project, pom.xmlDependency.dependency) != null
        } else {
            false
        }
    }
}
